Description: Fix certificate validation with QT 4.8. For some reason, the
 new on-demand root cert loading is not working with mumble.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/mumble/+bug/928296

diff -Naur mumble-1.2.3/src/SSL.cpp mumble-1.2.3.new/src/SSL.cpp
--- mumble-1.2.3/src/SSL.cpp	2011-02-19 22:35:16.000000000 +0100
+++ mumble-1.2.3.new/src/SSL.cpp	2012-02-16 02:45:20.353665502 +0100
@@ -252,6 +252,11 @@
 	QSslSocket::setDefaultCaCertificates(ql);
 #endif // NO_SYSTEM_CA_OVERRIDE
 
+#if QT_VERSION >= 0x040800
+	// Don't perform on-demand loading of root certificates
+	QSslSocket::setDefaultCaCertificates(QSslSocket::systemCaCertificates());
+#endif
+
 	for (unsigned int i=0;i<sizeof(recommended_cas)/sizeof(recommended_cas[0]);++i) {
 		QSslCertificate cert(recommended_cas[i]);
 		if (! QSslSocket::defaultCaCertificates().contains(cert)) {
